File: authentication-action-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__AuthenticationActionDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.authenticationaction.AuthenticationAction;
import se.curity.identityserver.sdk.plugin.descriptor.AuthenticationActionPluginDescriptor;

public final class ${pluginName}AuthenticationActionDescriptor implements AuthenticationActionPluginDescriptor<${pluginName}AuthenticationActionConfig> {

    @Override
    public Class<? extends AuthenticationAction> getAuthenticationAction()
    {
        return ${pluginName}AuthenticationAction.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}AuthenticationActionConfig> getConfigurationType()
    {
        return ${pluginName}AuthenticationActionConfig.class;
    }    
}

File: kotlin-authentication-action-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__AuthenticationActionDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.AuthenticationActionPluginDescriptor

class ${pluginName}AuthenticationActionDescriptor: AuthenticationActionPluginDescriptor<${pluginName}AuthenticationActionConfig>
{
    override fun getAuthenticationAction() = ${pluginName}AuthenticationAction::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}AuthenticationActionConfig::class.java;
}
